### ISSUE TITLE
Why the weights are not intialized ?

### ISSUE THREAD

**Issue by lemo2012** [huggingface/pytorch-pretrained-BERT#339]
03/03/2019 14:13:01 - INFO - pytorch_pretrained_bert.modeling -   Weights of BertForMultiLabelSequenceClassification not initialized from pretrained model: ['classifier.weight', 'classifier.bias']
03/03/2019 14:13:01 - INFO - pytorch_pretrained_bert.modeling -   Weights from pretrained model not used in BertForMultiLabelSequenceClassification: ['cls.predictions.bias', 'cls.predictions.transform.dense.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.transform.LayerNorm.bias', 'cls.predictions.decoder.weight', 'cls.seq_relationship.weight', 'cls.seq_relationship.bias']

**Comment by jplehmann** [comment]
I was wondering this myself.  It looks like there's some configuration mismatch -- some parameters found which aren't used, and a few expected that aren't found.

I'm not sure if this is expected, since the top-level task-specific classifier is correctly NOT pre-trained... or if it's something more.

Furthermore, I don't see the MRPC accuracy 84-88% as expected.  Using  different seeds (indicated by the filename) I get either 80 or 84, which is not only lower than expected but also a weird distribution.
```
mrpc_output15/eval_results.txt
eval_accuracy = 0.803921568627451
eval_loss = 0.44585930088571474
global_step = 345
loss = 0.42245775305706523

mrpc_output16/eval_results.txt
eval_accuracy = 0.8455882352941176
eval_loss = 0.38226841594658645
global_step = 345
loss = 0.25925399116847825

mrpc_output17/eval_results.txt
eval_accuracy = 0.7916666666666666
eval_loss = 0.4917685123635273
global_step = 345
loss = 0.24811905570652174

mrpc_output3/eval_results.txt
eval_accuracy = 0.8431372549019608
eval_loss = 0.42019053533965467
global_step = 345
loss = 0.2503709876019022

mrpc_output42/eval_results.txt
eval_accuracy = 0.8406862745098039
eval_loss = 0.44909875124108556
global_step = 345
loss = 0.21309310249660326

mrpc_output44/eval_results.txt
eval_accuracy = 0.8406862745098039
eval_loss = 0.45059946084431574
global_step = 345
loss = 0.10150747223068839

mrpc_output45/eval_results.txt
eval_accuracy = 0.8063725490196079
eval_loss = 0.42597512491777834
global_step = 345
loss = 0.29104428498641305
```

**Comment by jplehmann** [comment]
I dug up some related issues which confirms my guess above -- this kind of message is expected since the models are not yet find-tuned to the task.

https://github.com/huggingface/pytorch-pretrained-BERT/issues/161
https://github.com/huggingface/pytorch-pretrained-BERT/issues/180

Now my only question is why I am seeing these lower scores.

**Comment by thomwolf** [comment]
Yes this is the expected behavior.
I don't want to make the warning messages says this is "all good" because in some case, depending on the model you are loading in, this could be an unwanted behavior (not loading all the weights).

**Comment by viva2202** [comment]
I continued pre-training with bert-base-uncased without fine tuning on round about 22K batches and the precision @ K for MaskedLM task did not change at all. Is the result legitimate or do I rather have a problem loading the weights? I received the same error message. The data set is from the automotive domain. At what point can I expect the weights to change?  Thank you very much for experience values.

**Comment by RafaelBMR** [comment]
@viva2202, I did the same here using directly the "run_language_modeling.py" script, but with 11k sequences (I continued pretraining using training data only), and then fine-tuned it using BertForSequenceClassification. Got 1.75% increase in accuracy compared to not continuing pretraining.
